.github/workflows/require-qc-label.yml

```diff
@@ -0,0 +1,31 @@
+name: QC Confirmation Required
+
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize, labeled, unlabeled]
+
+jobs:
+  enforce-qc-label:
+    name: QC Confirmation Check
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    steps:
+      - name: Check for 'qc-reviewed ✅' label
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ISSUE_URL: ${{ github.event.pull_request.issue_url }}
+        run: |
+          echo "🔍 Checking if 'qc-reviewed ✅' label is present..."
+
+          LABELS=$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" "$ISSUE_URL/labels" | jq -r '.[].name')
+
+          echo "Labels found: $LABELS"
+          if echo "$LABELS" | grep -q "qc-reviewed ✅"; then
+            echo "✅ Label found. QC check confirmed."
+            exit 0
+          else
+            echo "⚠️ Label 'qc-reviewed ✅' is missing. Please add it after reviewing QC reports."
+            echo "ℹ️ This check will block merge until the label is applied."
+            exit 78  # 'neutral' exit code
+          fi
```

Comment:
Darn, I was hoping that the neutral exit code would still be supported, but alas:

- https://github.com/orgs/community/discussions/15452